pkg/pipeline/common.go

```diff
@@ -0,0 +1,44 @@
+// pipeline contains pipelines that define all the steps that a CRD has to go
+// through in order to be considered reconciled.
+
+package pipeline
+
+const (
+	protectionSettingEnvVar    = "LIEUTENANT_DELETE_PROTECTION"
+	DeleteProtectionAnnotation = "syn.tools/protected-delete"
+)
+
+func common(obj PipelineObject, data *ExecutionContext) ExecutionResult {
+
+	result := handleDeletion(obj.GetObjectMeta(), data)
+	if result.Abort || result.Err != nil {
+		result.Err = wrapError("deletion", result.Err)
+		return result
+	}
+
+	result = addTenantLabel(obj, data)
```

Comment:
Is this really a common step? Currently the tenants themselves don't have a tenant label. It should probably happen together with `setTenantOwner()`?

Reply:
Right, I'll move it to gitrepo and cluster specific steps.